apps/api/src/app/admin/admin.service.ts

```diff
@@ -35,6 +37,38 @@ export class AdminService {
     this.baseCurrency = this.configurationService.get('BASE_CURRENCY');
   }
 
+  public async addAssetProfile({
+    dataSource,
+    symbol
+  }: UniqueAsset): Promise<SymbolProfile | never> {
+    try {
+      const assetProfile = await this.dataProviderService.getAssetProfiles([
+        { dataSource, symbol }
+      ]);
+
+      if (!assetProfile[symbol].currency) {
```

Comment:
Why do you check for `currency`?

```
if (!assetProfiles[symbol]) {
```

Reply:
I am checking for currency because the function returns the below son even when it does not get any data.
{
symbolName:{
datasourse: "providedDatasource"
symbol: "symbolName",
currency: null
}
}

Reply:
Ok